src/test/java/com/library/controller/BookControllerTest.java

```diff
@@ -0,0 +1,208 @@
+        assertEquals(LocalDateTime.now().format(DateTimeFormatter.ofPattern("dd-MM-yyyy hh:mm:ss")), jsonObject.get("timestamp"));
+    }
+
+    @SneakyThrows
+    @Test
+    void saveBook() {
+        Book newBook = testStorage.getNewBook();
+        BookDto bookDto = bookMapper.mapToBookDto(newBook);
+        newBook.setPublisher(testStorage.getPublisher());
+        newBook.setAuthors(testStorage.getAuthors());
+
+        when(bookRepository.save(argThat(new IsSameLikeBook(newBook)))).thenReturn(newBook);
+
+        MockHttpServletResponse response = mockMvc.perform(post("/books")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(bookDto)))
+                .andExpect(status().isOk())
+                .andReturn()
+                .getResponse();
+
+        JSONObject jsonObject = new JSONObject(response.getContentAsString());
+
+        assertNotNull(response);
+        assertEquals(testStorage.getBook().getTitle(), jsonObject.get("title"));
+        verify(bookRepository).save(argThat(new IsSameLikeBook(newBook)));
+    }
+
+    @SneakyThrows
+    @Test
+    void saveBookFailed() {
+        Book newBook = testStorage.getNewBook();
+        newBook.setAuthors(null);
+        BookDto bookDto = bookMapper.mapToBookDto(newBook);
+
+
+        MockHttpServletResponse response = mockMvc.perform(post("/books")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(bookDto)))
+                .andExpect(status().isBadRequest())
+                .andReturn()
+                .getResponse();
+
+        JSONObject jsonObject = new JSONObject(response.getContentAsString());
+
+        assertNotNull(response);
+        assertEquals(MISSING_TYPE, jsonObject.get("message"));
+    }
+
+    @SneakyThrows
+    @Test
+    void updateBook() {
+        Book book = testStorage.getBook();
+        book.setTitle("update");
+        BookDto bookDto = bookMapper.mapToBookDto(book);
+
+        when(bookRepository.getReferenceById(book.getId())).thenReturn(book);
+        when(bookRepository.save(argThat(new IsSameLikeBook(book)))).thenReturn(book);
+
+        MockHttpServletResponse response = mockMvc.perform(put("/books")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(bookDto)))
+                .andExpect(status().isOk())
+                .andReturn()
+                .getResponse();
+
+        JSONObject jsonObject = new JSONObject(response.getContentAsString());
+
+        assertNotNull(response);
+        assertEquals(book.getTitle(), jsonObject.get("title"));
+        verify(bookRepository).save(argThat(new IsSameLikeBook(book)));
+        verify(bookRepository).getReferenceById(book.getId());
+    }
+
+    @SneakyThrows
+    @Test
+    void softDeleteBook() {
```

Comment:
Test without repository method mock?

Did you run tests without starting postgres container?

Reply:
With running container, but repository was mocked... data from DB wasn't used. I added method mock and verify to test method but nothing has changed

Reply:
"Did you run tests without starting postgres container?"

Where is the answer on this question?

Reply:
"With running container" is answer)